services/github.py

```diff
@@ -24,3 +27,35 @@ def get_github_integration_token(
     except InvalidInstallationError:
         log.warning("Failed to get installation token")
         raise RepositoryWithoutValidBotError()
+
+
+COMMIT_GHAPP_KEY_NAME = lambda commit_id: f"app_to_use_for_commit_{commit_id}"
+
+
+def set_github_app_for_commit(
+    installation_id: str | int | None, commit: Commit
```

Comment:
Hmm, `installation_id` has a lot of different  types. In what circumstances should we be expecting a `str` vs` int` vs `None` type? Should this be documented?